omod/src/main/java/org/openmrs/module/appui/rest/SessionController.java

```diff
@@ -47,6 +51,21 @@ public Object get(HttpServletRequest request, HttpServletResponse response) {
         return ret;
     }
 
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseBody
+    public Object set(@RequestBody Map<String, Object> requestBody,
+                      HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        if(requestBody.get("location") != null){
+            UiSessionContext uiSessionContext = new UiSessionContext(locationService, providerService, request);
+            Location location = locationService.getLocationByUuid(requestBody.get("location").toString());
```

Comment:
If the passed uuid is for a non existing location, what happens for .toString()?